js/src/components/contentAnalysis/mapResults.js

```diff
@@ -0,0 +1,97 @@
+import scoreToRating from "yoastseo/js/interpreters/scoreToRating";
+
+/**
+ * Mapped result definition.
+ * @typedef {Object} MappedResult
+ * @property {string} rating
+ * @property {bool} hasMarks
+ * @property {string} text
+ * @property {string} id
+ * @property {func} marker
+ * @property {number} score
+ */
+
+/**
+ * Mapped results definition.
+ * @typedef {Object} MappedResults
+ * @property {Array<MappedResult>} errorsResults
+ * @property {Array<MappedResult>} problemsResults
+ * @property {Array<MappedResult>} improvementsResults
+ * @property {Array<MappedResult>} goodResults
+ * @property {Array<MappedResult>} considerationsResults
+ */
+
+/**
+ * Maps a single results to a result that can be interpreted by yoast-component's ContentAnalysis.
+ *
+ * @param {object} result Result provided by YoastSEO.js.
+ *
+ * @returns {MappedResult} The mapped result.
+ */
+function mapResult( result ) {
+	const mappedResult = {
+		score: result.score,
+		rating: scoreToRating( result.score ),
+		hasMarks: result._hasMarks,
```

Comment:
`_` means a property is private. We should be using functions that access these. If there are no functions that retrieve those values, they should be created.